src/lib/protocols/mdns/Publisher.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "Publisher.h"
+
+#include <inttypes.h>
+
+#include "lib/protocols/mdns/platform/Mdns.h"
+#include "lib/support/logging/CHIPLogging.h"
+#include "platform/CHIPDeviceBuildConfig.h"
+#include "platform/CHIPDeviceLayer.h"
+#include "support/CodeUtils.h"
+#include "support/ErrorStr.h"
+
+namespace chip {
+namespace Protocols {
+namespace Mdns {
+
+CHIP_ERROR Publisher::Init()
+{
+    return ChipMdnsInit(HandleMdnsInit, HandleMdnsError, this);
```

Comment:
this implies that Publisher can be the only client of ChipMdns?

Reply:
Yes. We can add new methods to this class for future feature requests.

Reply:
which class?

Reply:
The `Publisher` class. I plan to handle all the service registration things here.